climada/engine/impact.py

```diff
@@ -862,6 +865,77 @@ def write_col(i_col, imp_ws, xls_data):
 
         imp_wb.close()
 
+    def write_hdf5(self, file_data, todense=False):
+        """Write centroids attributes into hdf5 format.
+
+        Parameters
+        ----------
+        file_data : str or h5
+            If string, path to write data. If h5 object, the datasets will be generated there.
+        todense: bool
+            if True write the sparse matrices as hdf5.dataset by converting them to dense format
+            first. This increases readability of the file for other programs. default: False
+        """
+        if isinstance(file_data, (str, Path)):
+            LOGGER.info('Writing %s', file_data)
+            with h5py.File(file_data, 'w') as data:
+                self._write_hdf5(data, todense)
+        else:
+            self._write_hdf5(file_data, todense)
+
+    def _write_hdf5(self, data, todense):
+        str_dt = h5py.special_dtype(vlen=str)
+        for (var_name, var_val) in self.__dict__.items():
+            if var_name == 'tag':
+                # write the dictionary of tags of exposures, impact functions set and hazard
+                # write hazard tag
+                hf_str = data.create_dataset('haz_tag_haz_type', (1,), dtype=str_dt)
+                hf_str[0] = var_val['haz'].haz_type
```

Comment:
I think [Attributes](https://docs.h5py.org/en/stable/high/attr.html) are much more suitable to store single-value string (meta-)data. They are created automatically through a dictionary-type lookup:
```suggestion
                data.attrs["haz_tag_haz_type"] = var_val["haz"].haz_type
```
(Same goes for all other stored strings). See also https://docs.h5py.org/en/stable/quick.html#attributes

Reply:
Good suggestion and thanks for implementing.